ganja.js

```diff
@@ -1454,7 +1454,7 @@
           /^\d+[.]{0,1}\d*[E][+-]{0,1}\d*|^\.\d+[E][+-]{0,1}\d*|^0x\d+|^\d+[.]{0,1}\d*|^\.\d+|^\(\/.*[^\\]\/\)/g,                       // 3: literal hex, nonsci numbers and regex (surround regex with extra brackets!)
           /^(\.Normalized|\.Length|\.\.\.|>>>=|===|!==|>>>|<<=|>>=|=>|\|\||[<>\+\-\*%&|^\/!\=]=|\*\*|\+\+|\-\-|<<|>>|\&\&|\^\^|^[{}()\[\];.,<>\+\-\*%|&^!~?:=\/]{1})/g,   // 4: punctuator
           /^[A-Za-z0-9_]*/g]                                                                                                            // 5: identifier
```

Comment:
My guess is that the actual bug is on this line, and this should be something more like the regex in [this post](https://stackoverflow.com/a/6926184/102441),

```suggestion
          /^[$_\p{L}][$_\p{L}\p{Mn}\p{Mc}\p{Nd}\p{Pc}\u200C\u200D]*+/gu]                                                                                                            // 5: identifier
```

I have neither tested this nor checked that the answer from that post is correct.

Reply:
Hey @eric-wieser, thanks a lot for your review and code suggestion! I only had to remove the one-or-more `+` at the end of your unicode regex and it works great for everything I tested on it, like:

```js
"ℇℇasöäøæ09æøopüßdπ$π".match(/^[$_\p{L}][$_\p{L}\p{Mn}\p{Mc}\p{Nd}\p{Pc}\u200C\u200D]*/gu)
```

![image](https://user-images.githubusercontent.com/5236548/102706508-d3bd7780-4292-11eb-9c93-4fe0a92c93dd.png)

I guess every browser should support unicode regexes nowadays, so that shouldn't be a problem.